src/module.c

```diff
@@ -6704,6 +6735,130 @@ int RM_SetModuleUserACL(RedisModuleUser *user, const char* acl) {
     return ACLSetUser(user->user, acl, -1);
 }
 
+/* Retrieve the user ID of the client connection behind the current context.
+ * The user ID can be used later, in order to check if command, key or channel
+ * can be executed or accessed according to the ACLs rules associated with that user.
+ * The caller can later free the user ID using the function RM_FreeUserID
+ *
+ * On success a RedisModuleUserID object is returned, otherwise
+ * NULL is returned. */
+ RedisModuleUserID *RM_GetCurrentUserID(RedisModuleCtx *ctx) {
+    if (ctx->client->user == NULL) {
+        return NULL;
+    }
+
+    RedisModuleUserID *userid = zmalloc(sizeof(RedisModuleUserID));
+    userid->id = createStringObject(ctx->client->user->name, sdslen(ctx->client->user->name));
+    return userid;
+}
+
+/* Frees a given user ID .*/
+void RM_FreeUserID(RedisModuleUserID *userid) {
+    decrRefCount(userid->id);
+    zfree(userid);
+}
+
+/* Checks if the command can be executed by the user, identified by the specified user ID,
+ * according to the ACLs associated with it.
+ *
+ * On success a REDISMODULE_OK is returned, otherwise
+ * REDISMODULE_ERR is returned and errno is set to the following values:
+ *
+ * * ENOENT: Specified command does not exist.
+ * * ENOTSUP: No ACL user for the specified User ID
+ * * EACCES: Command cannot be executed, according to ACL rules
+ */
+int RM_ACLCheckCommandPermissions(RedisModuleUserID *userid, RedisModuleString **argv, int argc) {
+    int keyidxptr;
+    struct redisCommand *cmd;
+
+    /* Find command */
+    if ((cmd = lookupCommand(argv[0]->ptr)) == NULL) {
+        errno = ENOENT;
+        return REDISMODULE_ERR;
+    }
+
+    /* Get ACL user from RedisModuleUserID */
+    user *acl_user = ACLGetUserByName(userid->id->ptr, sdslen(userid->id->ptr));
+    if (acl_user == NULL) {
+        errno = ENOTSUP;
+        return REDISMODULE_ERR;
+    }
+
+    if (ACLCheckAllUserCommandPerm(acl_user, cmd, argv, argc, &keyidxptr) != ACL_OK) {
+        errno = EACCES;
+        return REDISMODULE_ERR;
+    }
+
+    return REDISMODULE_OK;
+}
+
+/* Check if the key can be accessed by the user, identified by the specified user ID,
+ * according to the ACLs associated with it.
+ *
+ * If the user can access the key, REDISMODULE_OK is returned, otherwise
+ * REDISMODULE_ERR is returned and errno is set to the following values:
+ * * ENOTSUP: No ACL user for the specified User ID
+ * * EACCES: Key cannot be accessed according to ACL rules
+ */
+int RM_ACLCheckKeyPermissions(RedisModuleUserID *userid, RedisModuleString *key) {
+    /* Get ACL user from RedisModuleUserID */
+    user *acl_user = ACLGetUserByName(userid->id->ptr, sdslen(userid->id->ptr));
```

Comment:
Module created users will fail here, since they are created outside of the regular ACL system. I believe you can still pull it out of the user successfully though, since the code in `RM_GetCurrentUserID` just grabs the name of the currently attached user. It doesn't look like there is any test validating this behavior.